### PR TITLE
Refactor to only support tokens and older versions of Jupyter Hub

### PR DESCRIPTION
**Pull request by DonJayamanne**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,7 +91,6 @@
         "@jupyterlab/services": "^7.0.5",
         "@vscode/extension-telemetry": "^0.7.7",
         "buffer": "^6.0.3",
-        "cookie": "^0.5.0",
```
Comment (DonJayamanne) [inline package.json:94]: No more cookies

File: src/authenticators/authenticator.ts [hunk 2]
```diff
@@ -1,309 +1,55 @@
...
-            };
-        }
-        if (this.useTokensForAuth) {
-            // This is the preferred and default.
-            return this.generateApiToken(options, token);
-        }
-        const cookieStore = new this.CookieStore();
-        await this.getBaseAuthInfo(options, cookieStore, token);
-        const jupyterHubUrl = getJupyterUrl(options.baseUrl, options.authInfo.username);
-        this.trackLogoutUrl(options.authInfo.username, options.baseUrl, cookieStore);
```
Comment (DonJayamanne) [inline src/authenticators/authenticator.ts:67]: No more cookies

File: src/common/cookieStore.base.ts [hunk 3]
```diff
@@ -1,28 +0,0 @@
-// Copyright (c) Microsoft Corporation.
-// Licensed under the MIT License.
-
-export abstract class BaseCookieStore {
```
Comment (DonJayamanne) [inline src/common/cookieStore.base.ts:4]: No more cookies

File: src/common/crypto.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+let cryptoProvider: Crypto;
+declare var WorkerGlobalScope: Function | undefined;
+// Web
+if (typeof window === 'object') {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    cryptoProvider = (window as any).crypto;
+}
+// Web worker
+else if (typeof WorkerGlobalScope !== 'undefined' && self instanceof WorkerGlobalScope) {
```
Comment (DonJayamanne) [inline src/common/crypto.ts:15]: For telemetry

File: src/common/telemetry.ts [hunk 5]
```diff
@@ -71,3 +71,11 @@ export function publicLog2<E extends ClassifiedEvent<OmitMetadata<T>> = never, T
         : disposableStore.add(new TelemetryReporter(AppInsightsKey));
     telemetryReporter.sendTelemetryEvent(eventName, data);
 }
+
+const urlsAndVersion = new Map<string, string>();
+export function sendJupyterHubVersion(baseUrl: string, version: string, _serverId: string) {
+    urlsAndVersion.set(baseUrl, version);
+}
+export function trackUsageOfOldApiGeneration(_baseUrl: string) {
+    //
```
Comment (DonJayamanne) [inline src/common/telemetry.ts:80]: Will fill this in a separate PR

File: src/test/suite/extension.test.ts [hunk 6]
```diff
@@ -10,120 +10,99 @@ import { noop } from '../../common/utils';
...
-    ].forEach(({ title, password, isApiToken }) => {
+        { title: 'password', password: () => 'pwd' }, // Test with password.
+        { title: 'token', password: () => apiToken } // Test with a user entered API token.
+    ].forEach(({ title, password }) => {
         describe(title, function () {
-            before(function () {
-                if (isWebExtension() && password() === hubToken) {
-                    // Web does not support tokens generated via CLI.
-                    // API tokens must be generated via the REST API using username/password.
-                    return this.skip();
```
Comment (DonJayamanne) [inline src/test/suite/extension.test.ts:68]: Re-enabled the tests

**Review by rebornix — DISMISSED** [review]

**Review by bpasero — APPROVED** [review]

**Pull request merged**